src/main/java/net/coderbot/iris/mixin/MixinGameRenderer.java

```diff
@@ -24,4 +34,17 @@
 		Iris.logger.info("GPU: " + GlUtil.getRenderer() + " (Supports OpenGL " + GlUtil.getOpenGLVersion() + ")");
 		Iris.logger.info("OS: " + System.getProperty("os.name"));
 	}
+
+	@Inject(method = "getProjectionMatrix", at = @At(value = "INVOKE", target = "Lcom/mojang/math/Matrix4f;multiply(Lcom/mojang/math/Matrix4f;)V"), locals = LocalCapture.CAPTURE_FAILHARD)
+    private void scaleHandDepth(Camera camera, float f, boolean bl, CallbackInfoReturnable<Matrix4f> cir, PoseStack poseStack) {
+        if (HandRenderer.isActive()) {
+            // This value is taken directly from Shaders Mod.
+            poseStack.scale(1F, 1F, 0.125F);
+        }
+    }
+
+	@Redirect(method = "renderLevel", at = @At(value = "FIELD", target = "Lnet/minecraft/client/renderer/GameRenderer;renderHand:Z"))
+	private boolean disableVanillaHandRendering(GameRenderer gameRenderer) {
+		return false;
```

Comment:
We probably don't want to touch vanilla hand rendering when shaders are disabled for any potential mod compatibility issues.

Reply:
Done.